### PR TITLE
Make sure to remove temp folder of cp2fm

### LINKED ISSUE #76: tmp folder not cleaned up after run
With every execution of `aemanalyser-maven-plugin:project-analyse` in version 1.0.4 there is three new folders below the Java tmp folder generated: one with prefix `cp2fm-converter` and two folders starting with prefix `slingfeature`. All three have a random number after the prefix.
All three together have a substantial size (675MB). Each new build adds another three folders.

Please make sure to cleanup the java tmp folder so that there are no leftovers after the Maven build.

### PR DESCRIPTION
**Pull request by kwin**
This closes #76

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by kwin** [comment]
This fixes the dangling big `cp2fm-converter` folders but not the empty `slingfeature` ones. The latter probably require a fix in cp2fm.

File: aemanalyser-maven-plugin/src/main/java/com/adobe/aem/analyser/AemPackageConverter.java [hunk 1]
```diff
@@ -116,13 +116,18 @@ public void convert(final Map<String, File> contentPackages) throws IOException
...
+                logger.info("Converting package {}", entry.getKey());
+                try {
+                    converter.convert(entry.getValue());
+                } catch (final Throwable t) {
+                    throw new IOException("Content Package Converter Exception " + t.getMessage(), t);        
+                }
             }
+        } finally {
+            // make sure to remove the temp folders
+            converter.cleanup();
```
Comment (henrykuijpers) [inline aemanalyser-maven-plugin/src/main/java/com/adobe/aem/analyser/AemPackageConverter.java:130]: If the assumption is that the temp folders can always be deleted, it should be treated as an error if that fails. Therefore, I propose to let `cleanup` throw `IOException`, that can be handled (or bubbled through to the calling code).

It also goes in line with the suggestion of making AemPackageConverter Closeable, as it is able to throw `IOException`.

Comment (kwin) [reply aemanalyser-maven-plugin/src/main/java/com/adobe/aem/analyser/AemPackageConverter.java:130]: https://github.com/apache/sling-org-apache-sling-feature-cpconverter/pull/86

**Comment by kwin** [comment]
@bosschaert Any chance you can spin a new release containing this fix?

**Comment by cziegeler** [comment]
The PR looks in general ok to me, although I still find the two methods (getLocalAP, getAP) confusing. I think that could be a single method. Regardless, one seems now to return an ArtifactManager so that should be renamed

**Comment by kwin** [comment]
Using a single method does not really work, because the ArtifactManager has to be released/closed while the other one only implements `ArtifactProvider` and hence doesn't need closing, but I agree that the method names are confusing. Any suggestions? I am not really understanding why the composition of the providers is not the other way round (i.e. ArtifactManager acting as the composite provider and the other one implementing `spi.ArtifactProvider` instead)

**Comment by kwin** [comment]
I cleaned up the method names in https://github.com/adobe/aemanalyser-maven-plugin/pull/77/commits/0f46a04fcb5b161bd255b28d4c975d553472eae3. @cziegeler Please have another look.

**Review by cziegeler — APPROVED** [review]
Lgtm, thanks

**Pull request merged**